src/libtorch.cc

```diff
@@ -0,0 +1,1122 @@
+        (std::string("skipping model configuration auto-complete for '") +
+         Name() + "': max_batch_size, inputs or outputs already specified")
+            .c_str());
+    return nullptr;  // success
+  }
+
+  std::string artifact_name;
+  RETURN_IF_ERROR(
+      ModelConfig().MemberAsString("default_model_filename", &artifact_name));
+
+  if (TRITONSERVER_LogIsEnabled(TRITONSERVER_LOG_VERBOSE)) {
+    triton::common::TritonJson::WriteBuffer buffer;
+    RETURN_IF_ERROR(ModelConfig().PrettyWrite(&buffer));
+    LOG_MESSAGE(
+        TRITONSERVER_LOG_INFO,
+        (std::string("post auto-complete:\n") + buffer.Contents()).c_str());
+  }
+
+  return nullptr;  // success
+}
+
+
+//
+// ModelInstanceState
+//
+// State associated with a model instance. An object of this class is
+// created and associated with each TRITONBACKEND_ModelInstance.
+//
+class ModelInstanceState : public BackendModelInstance {
+ public:
+  static TRITONSERVER_Error* Create(
+      ModelState* model_state,
+      TRITONBACKEND_ModelInstance* triton_model_instance,
+      ModelInstanceState** state);
+  virtual ~ModelInstanceState();
+
+  // Get the state of the model that corresponds to this instance.
+  ModelState* StateForModel() const { return model_state_; }
+
+  // Execute...
+  void ProcessRequests(
+      TRITONBACKEND_Request** requests, const uint32_t request_count);
+
+ private:
+  ModelInstanceState(
+      ModelState* model_state,
+      TRITONBACKEND_ModelInstance* triton_model_instance);
+  TRITONSERVER_Error* ValidateBooleanSequenceControl(
+      triton::common::TritonJson::Value& sequence_batching,
+      const std::string& control_kind, bool required, bool* have_control);
+  TRITONSERVER_Error* ValidateTypedSequenceControl(
+      triton::common::TritonJson::Value& sequence_batching,
+      const std::string& control_kind, bool required, bool* have_control);
+  TRITONSERVER_Error* ValidateInputs();
+  TRITONSERVER_Error* ValidateOutputs();
+  void Execute(
+      std::vector<TRITONBACKEND_Response*>* responses,
+      const uint32_t response_count,
+      std::vector<torch::jit::IValue>* input_tensors,
+      std::vector<torch::Tensor>* output_tensors);
+  void SetInputTensors(
+      size_t total_batch_size, TRITONBACKEND_Request** requests,
+      const uint32_t request_count,
+      std::vector<TRITONBACKEND_Response*>* responses,
+      BackendInputCollector* collector, std::vector<const char*>* input_names,
+      std::vector<torch::jit::IValue>* input_tensors, bool* cuda_copy);
+  void ReadOutputTensors(
+      size_t total_batch_size, const std::vector<const char*>& output_names,
+      const std::vector<torch::Tensor>& output_tensors,
+      TRITONBACKEND_Request** requests, const uint32_t request_count,
+      std::vector<TRITONBACKEND_Response*>* responses);
+
+  ModelState* model_state_;
+
+  // The full path to the TorchScript model file.
+  std::string model_path_;
+
+  std::shared_ptr<torch::jit::script::Module> torch_model_;
```

Comment:
I think the segfault is caused by torch_model_ being null.
Here I think you can use unique_ptr... there is no need to have multiple owners for torch_model_.

Reply:
I used a shared pointer because torch::jit::load() returns a shared pointer i.e. `std::shared_ptr<torch::jit::script::Module>` and not `torch::jit::script::Module`